### DISCUSSION TITLE
E2Function to get lambda of e2function

### DISCUSSION THREAD

**Discussion by Denneisk** [wiremod/wire#3036] [Suggestions]
Function that returns an extension function. For example,
```ts
let F = getFunction("print(ns)")

F(_HUD_PRINTCHAT, "test") #// Equivalent to print(_HUD_PRINTCHAT, "test")
```
This avoids having to wrap a function to use it as a lambda.

I know there's no provisions for varargs in lambdas yet, but we can just ban varargs until a solution comes around.

Is `getFunction` a good name? What about `e2function`?

**Comment by Denneisk** [comment]
@Vurv78 hoping you can comment on this also since I still respect your authority over the functions library.

**Reply by vurvdev** [reply]
I don't see a problem with this, no. Also you can / should implement varargs as just passing an array, if they provide a vararg signature

**Reply by vurvdev** [reply]
> I don't see a problem with this, no. Also you can / should implement varargs as just passing an array, if they provide a vararg signature

Was always intended to happen, this would probably be the best way to do it since overloads exist and adding a syntax to reference a specific overload would be pretty gross.

**Comment by Denneisk** [comment]
Only reason I haven't implemented this yet is because I kinda hate the syntax of it and disambiguation between overloads.

Right now I'm leaning towards something more like this:

```ts
let F = getFunction("print", "number", "string") # alternative 1, parameters as varargs
let F = getFunction("print", [ "number", "string" ]) # alternative 2, parameters as an optional array
```

And for ambiguous overloads, have `getFunction` return the least complex option. For `"print"` that would be `print(...)` > `print(ns)` since `print(...)` can be `print()`.

... I should just rewrite the type system while I'm at it, fmscl.

**Reply by Denneisk** [reply]
This syntax
```ts
let F = print<number, string>
```
might be best imo but ruins everything

**Reply by Denneisk** [reply]
Meditating on this has led me to the cusp of reinventing string calling.

**Reply by vurvdev** [reply]
I don't think there's any prior art in any language (haven't done research though) for referencing specific function overloads so I'd rather keep it simple and go with the `getFunction(name, args)` or `delegate(s,r)` whatever. That's what I was going to do.

**Reply by vurvdev** [reply]
Could also potentially make a string call just syntax sugar that uses the delegate() function too. That would be cool and ease the burden on having it in the compiler.
